src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/Snapshot/DependenciesSnapshot.cs

```diff
@@ -148,27 +155,42 @@ bool RemoveTargetFrameworksWithNoDependencies()
 
         public DependenciesSnapshot RemoveTargets(IEnumerable<ITargetFramework> targetToRemove)
         {
-            ImmutableDictionary<ITargetFramework, ITargetedDependenciesSnapshot> newTargets = Targets.RemoveRange(targetToRemove);
+            ImmutableDictionary<ITargetFramework, ITargetedDependenciesSnapshot> newTargets = DependenciesByTargetFramework.RemoveRange(targetToRemove);
 
             // Return this if no targets changed
-            return ReferenceEquals(newTargets, Targets)
+            return ReferenceEquals(newTargets, DependenciesByTargetFramework)
                 ? this
-                : new DependenciesSnapshot(ProjectPath, ActiveTarget, newTargets);
+                : new DependenciesSnapshot(ProjectPath, ActiveTargetFramework, newTargets);
         }
 
         // Internal, for test use -- normal code should use the factory methods
         internal DependenciesSnapshot(
             string projectPath,
-            ITargetFramework activeTarget,
-            ImmutableDictionary<ITargetFramework, ITargetedDependenciesSnapshot> targets)
+            ITargetFramework activeTargetFramework,
+            ImmutableDictionary<ITargetFramework, ITargetedDependenciesSnapshot> dependenciesByTargetFramework)
         {
             Requires.NotNullOrEmpty(projectPath, nameof(projectPath));
-            Requires.NotNull(activeTarget, nameof(activeTarget));
-            Requires.NotNull(targets, nameof(targets));
+            Requires.NotNull(activeTargetFramework, nameof(activeTargetFramework));
+            Requires.NotNull(dependenciesByTargetFramework, nameof(dependenciesByTargetFramework));
+
+            if (activeTargetFramework.Equals(TargetFramework.Empty))
+            {
+                if (dependenciesByTargetFramework.Count != 0)
```

Comment:
Requires.Argument with the condition would simplify these checks.

Reply:
Done.

I considered this at the time but because the validation is on the relationship between two arguments it wasn't clear which one to blame.

Reply:
For the BCL, we typically just picked the first one in the relationship.